salt/utils/decorators/__init__.py

```diff
@@ -245,7 +245,7 @@ def memoize(func):
 
     @wraps(func)
     def _memoize(*args, **kwargs):
-        args_ = ','.join(list(args) + ['{0}={1}'.format(k, kwargs[k]) for k in sorted(kwargs)])
+        args_ = ','.join([str(arg) for arg in args] + ['{0}={1}'.format(str(k), str(kwargs[k])) for k in sorted(kwargs)])
```

Comment:
The problem with just using `str()` in PY2 is that it causes a ``UnicodeDecodeError`` when you do it on a unicode string:

```python
>>> x = [1, 2, 3, u'✓']
>>> x
[1, 2, 3, u'\u2713']
>>> y = [str(z) for z in x]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
UnicodeEncodeError: 'ascii' codec can't encode character u'\u2713' in position 0: ordinal not in range(128)
```

So, while your code is more concise, an approach like [this](https://github.com/saltstack/salt/blob/c7c5266/salt/utils/timed_subprocess.py#L43-L49) would be better here.